hogben/tests/test_fisher.py

```diff
@@ -0,0 +1,337 @@
+    np.testing.assert_allclose(g_reference, g_compare, rtol=1e-02, atol=0)
+
+
+@patch('hogben.utils.reflectivity')
+@pytest.mark.parametrize('model_class', ("mock_refl1d_model",
+                                         "mock_refnx_model"))
+@pytest.mark.parametrize('model_params', (1, 2, 3, 4))
+def test_fisher_shape(mock_reflectivity, model_params, model_class, request):
+    """
+    Tests whether the shape of the Fisher information matrix remains
+     correct when changing the amount of parameters
+    """
+    model = request.getfixturevalue(model_class)
+    xi = model.xi[:model_params]
+
+    mock_reflectivity.side_effect = get_mock_reflectivity()
+
+    expected_shape = (model_params, model_params)
+    g = get_fisher_information([model], xi=xi)
+    np.testing.assert_array_equal(g.shape, expected_shape)
+
+
+@patch('hogben.utils.reflectivity')
+@pytest.mark.parametrize('model_class', ("mock_refl1d_model",
+                                         "mock_refnx_model"))
+@pytest.mark.parametrize('qs',
+                         (np.arange(0.001, 1.0, 0.25),
+                          np.arange(0.001, 1.0, 0.10),
+                          np.arange(0.001, 1.0, 0.05),
+                          np.arange(0.001, 1.0, 0.01)))
+def test_fisher_diagonal_positive(mock_reflectivity, qs, model_class, request):
+    """Tests whether the diagonal values in the Fisher information matrix
+     are positively valued"""
+    model = request.getfixturevalue(model_class)
+    mock_reflectivity.side_effect = get_reflectivity_given_datapoints(len(qs))
+    g = get_fisher_information([model], qs=[qs])
+    np.testing.assert_array_less(np.zeros(len(g)), np.diag(g))
+
+
+@pytest.mark.parametrize('model_class', ("mock_refl1d_model",
+                                         "mock_refnx_model"))
+@pytest.mark.parametrize('model_params', (1, 2, 3, 4))
+def test_fisher_no_data(model_params, model_class, request):
+    """Tests whether a model with zero data points properly returns an empty
+    matrix of the correct shape"""
+    model = request.getfixturevalue(model_class)
+    xi = model.xi[:model_params]
+    qs = []
+    g = get_fisher_information([model], qs=[qs], xi=xi)
+    np.testing.assert_equal(g, np.zeros((len(xi), len(xi))))
+
+
+@pytest.mark.parametrize('model_class', ("mock_refl1d_model",
+                                         "mock_refnx_model"))
+@patch('hogben.utils.reflectivity')
+def test_fisher_no_parameters(mock_reflectivity, model_class, request):
+    """Tests whether a model without any parameters properly returns a
+    zero array"""
+    model = request.getfixturevalue(model_class)
+    mock_reflectivity.side_effect = get_mock_reflectivity()
+    g = get_fisher_information([model], xi=[])
+    np.testing.assert_equal(g.shape, (0, 0))
+
+
+@pytest.mark.parametrize('model_class', ("refnx_model",
+                                         "refl1d_model"))
+def test_multiple_identical_models(model_class, request):
```

Comment:
I think this could have a slightly more descriptive name, e.g. what we are testing is that the FIM scales when we repeat a "measurement"

Reply:
Renamed it to `test_fisher_doubling_with_two_identical_models`, as it tests that the FIM has double values when using two identical models.